apis/core/lib/pipeline/trigger.ts

```diff
@@ -0,0 +1,25 @@
+import { URLSearchParams } from 'url'
+import { NamedNode } from 'rdf-js'
+import fetch from 'node-fetch'
+import env from '@cube-creator/core/env'
+
+const pipelineURI = env.PIPELINE_URI
+
+export async function triggerPipeline(job: NamedNode): Promise<void> {
```

Comment:
Let's call this a `triggerLocalPipeline` or similar to make it clear already that it's used in local env

Reply:
In a perfect world we could do the switch via the url in env. Ok, I am not sure this will work

Reply:
This will not be a simple URL substitution. The HTTP request to trigger GitHub or GitLab will be completely different.

Reply:
What do you think of think of this approach

Reply:
Yes, I was thinking similar. Can be even simpler by star import and using the env var as function name:

```javascript
import * as triggers from './lib/pipeline/trigger'

const trigger = triggers[env.PIPELINE_TYPE]

if (!trigger) {
  throw new Error(`Trigger ${env.PIPELINE_TYPE} is not implemented`)
}

trigger(job)
```

and in `lib/pipeline/trigger.ts`

```js
export function local() {
}

export function github() {
}

export function gitlab() {
}
```

Reply:
Fancy 👍